### PR TITLE
Check for instance READY state before placing a share on it.

### PR DESCRIPTION
**Pull request by tyuchn**
**What type of PR is this?**
> Uncomment only one ` /kind <>` line, hit enter to put that in a new line, and remove leading whitespaces from that line:
>
> /kind api-change
> /kind bug
> /kind cleanup
> /kind design
> /kind documentation
> /kind failing-test
> /kind feature
> /kind flake

**What this PR does / why we need it**:

**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note

```

### CODE REVIEW

File: pkg/csi_driver/multishare_ops_manager.go [hunk 1]
```diff
@@ -324,6 +324,11 @@ func (m *MultishareOpsManager) runEligibleInstanceCheck(ctx context.Context, ins
 	var readyEligibleInstances []*file.MultishareInstance
 	nonReadyInstanceCount := 0
 	for _, instance := range instances {
+		if instance.State != "READY" {
```
Comment (saikat-royc) [inline pkg/csi_driver/multishare_ops_manager.go:327]: we should consider CREATING as a non ready instance too

File: pkg/csi_driver/multishare_ops_manager.go [hunk 2]
```diff
@@ -345,16 +359,6 @@ func (m *MultishareOpsManager) runEligibleInstanceCheck(ctx context.Context, ins
...
-		if op.Type == util.InstanceDelete {
-			continue
-		}
-
-		if instance.State == "DELETING" || instance.State == "ERROR" {
-			klog.V(5).Infof("Instance %s/%s/%s with state %s is not eligible", instance.Project, instance.Location, instance.Name, instance.State)
-			continue
-		}
-
 		klog.Infof("Instance %s/%s/%s is not ready with ongoing operation %s type %s", instance.Project, instance.Location, instance.Name, op.Id, op.Type.String())
```
Comment (saikat-royc) [inline pkg/csi_driver/multishare_ops_manager.go:362]: let's also log the state here

Comment (tyuchn) [reply pkg/csi_driver/multishare_ops_manager.go]: Done

File: pkg/csi_driver/multishare_ops_manager.go (hunk 1 above)
Comment (jingxu97) [reply pkg/csi_driver/multishare_ops_manager.go:335]: what if the instance is in the process of restarting? do we need any logic to handle this case specifically or "not ready" can work for this case too?

**Pull request merged**